ui/components/extensions/adapters/adapters.js

```diff
@@ -0,0 +1,175 @@
+import { useEffect, useState } from "react";
+import { withRouter } from "next/router";
+import { iconMedium, extensionStyles as styles } from "../../../css/icons.styles";
+import { Grid, Typography, IconButton, Switch, CloseIcon } from "@material-ui/core";
+import { withStyles } from "@material-ui/core/styles";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import { withSnackbar } from "notistack";
+import { updateProgress } from "../../../lib/store";
+import { INITIAL_GRID_SIZE } from "../../../utils/Enum";
+import { adaptersList } from "./constants";
+import changeAdapterState from '../../graphql/mutations/AdapterStatusMutation';
+
+const Adapters = ({ updateProgress, enqueueSnackbar, closeSnackbar, classes, meshAdapters, meshAdaptersts }) => {
+
+  // States.
+  const [availableAdapters, setAvailableAdapters] = useState(adaptersList);
+  const [mts, setMts] = useState(new Date());
+
+  // useEffects.
+  // TODO: This way of updating progress is really bad. Need to make this more elegant.
+  useEffect(() => {
+    updateProgress({ showProgress : true })
+  }, [])
+
+  useEffect(() => {
+    if (meshAdaptersts > mts) {
+      let currentAdaptersList = { ...adaptersList };
+      meshAdapters.forEach(element => {
+        const adapterId = element.name;
+        if (adapterId && currentAdaptersList[adapterId]) {
+          currentAdaptersList[adapterId].enabled = true;
+          currentAdaptersList[adapterId].url = element.adapter_location;
+        }
+      })
+      setAvailableAdapters(currentAdaptersList);
+      setMts(meshAdaptersts);
+      updateProgress({ showProgress : false })
+    }
+
+  }, [meshAdapters, meshAdaptersts])
+
+
+  // Handlers.
+  const handleAdapterDeployment = (payload, msg, selectedAdapter, adapterId) => {
+    updateProgress({ showProgress : true });
+
+    changeAdapterState((response, errors) => {
+      updateProgress({ showProgress : false });
+
+      if (errors !== undefined) {
+        // Toggle the switch to it's previous state if the request fails.
+        setAvailableAdapters({ ...availableAdapters, [adapterId] : { ...selectedAdapter, enabled : !selectedAdapter.enabled } });
+        handleError(msg);
+      }
+
+
+      enqueueSnackbar(`Adapter ${response.adapterStatus.toLowerCase()}`, {
+        opt : {
```

Comment:
[The enqueueSnackbar api](https://notistack.com/api-reference) expects set of options directly as the second param.

Reply:
Yes I am aware about this but weirdly enough, that is not working I don't know why it's giving me import issues that way (passing options directly). Although the same has been implemented in other places also.

Reply:
Would you share the errors and messages, please describe.

Reply:
This is the error basically.

<img width="1011" alt="image" src="https://github.com/meshery/meshery/assets/39329182/32c239b0-94d6-4738-8c14-60de1a1bbfb8">

Reply:
use:
```
import { SnackbarProvider, useSnackbar } from 'notistack';

in React function, use hooks like this:
  const { enqueueSnackbar, closeSnackbar } = useSnackbar()
  
  
Don't use enqueuesnackbar from props.
  
```

Reply:
use this guide: https://notistack.com/getting-started

Reply:
@Abhishek-kumar09 actually I am aware about this, and I have tried this also, but it gives the same error. Although not sure, but a thing to notice is - we are actually using an old version of the library. 

Current version -> 3.X.X  and we are using 1.X.X

Reply:
@Abhishek-kumar09 okay so I finally got the problem. So basically the problem was with the import that I was doing i.e,  `CloseIcon` from material ui. It turns out that I can't import it directly from '@material-ui/core' and have to explicitly import it from `@material-ui/icons/Close`. But anyways I have added the `useSnackbar` hook.